runtime/src/bank_forks.rs

```diff
@@ -174,6 +174,12 @@ impl BankForks {
         highest_confirmed_root: Option<Slot>,
     ) {
         let old_epoch = self.root_bank().epoch();
+        // After setting the root here, we have to ensure that no other banks < root that
+        // are frozen are concurrently replayed. This is because the flush cache logic in
```

Comment:
Frozen won't be being replayed right?

Reply:
ah meant *not* frozen, i.e. half finished banks that are on an unrooted path.